Automattic/jetpack
# Add/segmentation params to registration

This PR enables us to inform the segmentation parameters on the registration step

#### Changes proposed in this Pull Request:

* Adds `from` and `aff` parameters to the registration step

#### Jetpack product discussion


p9dueE-2ER-p2
1191179647901802-as-1200125136103890

#### Does this pull request change what data or activity we track or use?


It adds the possibility to identify sites registered by partners on the registration step (we used to do this only at the authorization step)

#### Testing instructions:






First of all, test if the registration flow is still working as expected.

Setting up for the following tests:

* sandbox your site
* on you sandbox, edit the `register_site` method, and add a debug to see the contents of the `$data` variable on`/mu-plugins/jetpack/class.jetpack-data.php` line 215

TIP: use `wp jetpack disconnect blog` to disconnect your blog in between all the tests

##### Testing the connection/register v4 endpoint

* Initiate the connection from the main wp-admin dashboard
* make sure you see the `from` parameter getting to your sandbox
* initiate the connection from the Jetpack dashboard
* make sure you see the `from` parameter getting to your sandbox

<img width="354" alt="Captura de Tela 2021-04-22 às 11 38 06" src="https://user-images.githubusercontent.com/971483/115777099-46e31180-a38b-11eb-812e-bf1b4bf29971.png">

##### Testing remote_register XMLRPC

See 2911f-pb/#plain

##### Testing affiliates

* `wp option update jetpack_affiliate_code something`
* `wp option update jetpack_partner_subsidiary_id something_else`
* Connect the site
* Make sure you see `from`, `aff` and `subsidiaryId` params getting to your sandbox

##### Testing non iframe connection flow

* add `define( 'JETPACK_SHOULD_NOT_USE_CONNECTION_IFRAME', true );` to your `wp-config.php`
* Connect
* make sure you see the `from` parameter getting to your sandbox

##### Testing partner provisioning CLI

The script doesn't seem to be used anymore. See p9dueE-2ER-p2#comment-4901
